### PR TITLE
Code for SoundManager added

### PR DESCRIPTION
**Pull request by Akai-Ryu**
@fkarg Die Architektur für den SoundManager muss entsprechend angepasst werden.

### CODE REVIEW

File: Singularity/Singularity/Sound/SoundManager.cs [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+namespace Singularity.Sound
+{
+    public class SoundManager
+    {
+
+        // Sound listener for 3D audio.
+        private readonly AudioListener mListener;
+        // The z coordinate for 3D audio.
+        private float mSoundPlaneDepth;
+        // Dictionary containing all the songs assigned to their names and sorted by phase of the game.
```
Comment (yvan674) [inline Singularity/Singularity/Sound/SoundManager.cs]: Dictionaries don't have a sort so what do you mean by sorted by phase?

File: Singularity/Singularity/Sound/SoundManager.cs [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+        /// <param name="name">The sound effect's name.</param>
+        /// <param name="x">x coordinate where the sound is supposed to be played.</param>
+        /// <param name="y">y coordinate where the sound is supposed to be played.</param>
+        /// <param name="volume">The supposed sound volume.</param>
+        /// <param name="pitch">The pitch modificator.</param>
+        /// <param name="isGlobal">Specifies whether the sound has a position.</param>
+        /// <param name="loop">Specifies whether the sound is supposed to loop.</param>
+        /// <param name="soundClass">Specifies the SoundClass e.g. Effect.</param>
+        /// <returns>The id of the instance if params are valid. -1 if the SoundClass is invalid. -2 if the effect doesn't exist.</returns>
+        public int PlaySound(string name, float x, float y, float volume, float pitch, bool isGlobal, bool loop, SoundClass soundClass)
```
Comment (yvan674) [inline Singularity/Singularity/Sound/SoundManager.cs]: if the method for playing other sounds are static, should this also be made static?

File: Singularity/Singularity/Sound/SoundManager.cs [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+                return -1;
+            }
+        }
+
+        /// <summary>
+        /// Sets the volume of the sound effect instance specified by the given id.
+        /// </summary>
+        /// <param name="id">The global sound instance id.</param>
+        /// <param name="volume">The desired volume.</param>
+        public void SetSoundVolume(int id, float volume)
```
Comment (yvan674) [inline Singularity/Singularity/Sound/SoundManager.cs]: same thing regarding static

File: Singularity/Singularity/Sound/SoundManager.cs (hunk 1 above)
Comment (Akai-Ryu) [reply Singularity/Singularity/Sound/SoundManager.cs]: True, what I meant was "paired with". I changed it.

**Comment by fkarg** [comment]
@yvan674 In this case (when you approve of the changes) please do actually Approve of the changes in your review, since otherwise it has to be dismissed (which it can always be done but is of course not a good course of action)

File: Singularity/Singularity/Sound/SoundManager.cs [hunk 4]
```diff
@@ -0,0 +1,482 @@
...
+            mListener.Position = new Vector3(x, y, mSoundPlaneDepth);
+        }
+
+        /// <summary>
+        /// Change the position of a 3D sound effect instance.
+        /// </summary>
+        /// <param name="id">The gloabl id of the sound effect instance.</param>
+        /// <param name="x">The new x coordinate.</param>
+        /// <param name="y">The new y coordinate.</param>
+		public void SetSoundPosition(int id, float x, float y)
```
Comment (fkarg) [inline Singularity/Singularity/Sound/SoundManager.cs]: The indentation seems off here ... at least in the webview, can you look into that?

File: Singularity/Singularity/Sound/SoundManager.cs [hunk 5]
```diff
@@ -0,0 +1,482 @@
...
+        /// <param name="name">The level name.</param>
+		public void SetLevelThemeMusic(string name)
+        {
+            mLevel = name;
+        }
+
+        /// <summary>
+        /// To be called within a level or menu. Deletes all the stopped (disposable) instances.
+        /// </summary>
+	    public void CleanUpStoppedInstances()
```
Comment (fkarg) [inline Singularity/Singularity/Sound/SoundManager.cs]: weird indentation here again ...

File: Singularity/Singularity/Sound/SoundManager.cs [hunk 6]
```diff
@@ -0,0 +1,482 @@
...
+                // TODO: Maybe play random background music.
+            }
+            MediaPlayer.Play(mAllSongs[mLevel][(int)soundPhase]);
+        }
+
+        /// <summary>
+        /// Setting the theme song for the current level by specifying the current level. String.Empty() for random music from pool.
+        /// </summary>
+        /// <param name="name">The level name.</param>
+		public void SetLevelThemeMusic(string name)
```
Comment (fkarg) [inline Singularity/Singularity/Sound/SoundManager.cs]: oh and here

File: Singularity/Singularity/Sound/SoundManager.cs [hunk 7]
```diff
@@ -0,0 +1,482 @@
...
+            mEffectInstanceId = 0;
+            mUiInstanceId = 0;
+        }
+
+
+        /// <summary>
+        /// Set the z coordinate for all 3D sounds (effectivly the distance of the sound plane from the listener).
+        /// </summary>
+        /// <param name="z">The desired z coordinate.</param>
+	    public void SetSoundPlaneDepth(float z)
```
Comment (fkarg) [inline Singularity/Singularity/Sound/SoundManager.cs]: here as well

File: Singularity/Singularity/Sound/SoundManager.cs (hunk 4 above)
Comment (Akai-Ryu) [reply Singularity/Singularity/Sound/SoundManager.cs]: webview issue i guess

File: Singularity/Singularity/Sound/SoundManager.cs (hunk 5 above)
Comment (Akai-Ryu) [reply Singularity/Singularity/Sound/SoundManager.cs]: all fine locally

File: Singularity/Singularity/Sound/SoundManager.cs (hunk 6 above)
Comment (Akai-Ryu) [reply Singularity/Singularity/Sound/SoundManager.cs]: local version ok here

File: Singularity/Singularity/Sound/SoundManager.cs (hunk 7 above)
Comment (Akai-Ryu) [reply Singularity/Singularity/Sound/SoundManager.cs]: local version ok here

**Pull request merged**